source/test/sgp_mode_test/SGPModularity.test.cc

```diff
@@ -0,0 +1,389 @@
+#include "../../default_mode/Host.h"
+#include "../../default_mode/Symbiont.h"
+#include "../../sgp_mode/AnalysisTools.h"
+#include "../../sgp_mode/CPU.h"
+#include "../../sgp_mode/CPUState.h"
+#include "../../sgp_mode/GenomeLibrary.h"
+#include "../../sgp_mode/Instructions.h"
+#include "../../sgp_mode/ModularityAnalysis.h"
+#include "../../sgp_mode/SGPHost.h"
+#include "../../sgp_mode/SGPSymbiont.h"
+#include "../../sgp_mode/SGPWorld.h"
+#include "../../sgp_mode/Scheduler.h"
+#include "../../sgp_mode/Tasks.h"
+#include <iostream>
+#include <memory>
+
+// have world call these methods
+
+TEST_CASE("GetUsefulStarts", "[sgp]") {
+  GIVEN("A vector of 1s and 0s representing every site (instruction) in the "
+        "actual genome that is either"
+        "necessary to perform the designated task or not necessary to perform "
+        "the task, respectively") {
+    emp::vector<int> needed_code_sites_a = {1, 0, 0, 0, 0, 0, 1, 1, 1, 1,
+                                            0, 0, 0, 0, 0, 1, 1, 1, 0, 1};
+    emp::vector<int> needed_code_sites_b = {0, 0, 0, 0, 0, 0, 1, 1, 1, 1,
+                                            0, 0, 0, 0, 0, 1, 1, 1, 0, 1};
+    emp::vector<int> needed_code_sites_c = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                                            0, 0, 0, 0, 0, 1, 1, 1, 0, 1};
+    emp::vector<int> needed_code_sites_d = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                                            0, 0, 0, 0, 0, 0, 0, 0, 0, 1};
+    emp::vector<int> needed_code_sites_e = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                                            0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+
+    emp::vector<emp::vector<int>> needed_code_sites = {
+        needed_code_sites_a, needed_code_sites_b, needed_code_sites_c,
+        needed_code_sites_d, needed_code_sites_e};
+
+    emp::vector<int> calc_useful_starts = GetUsefulStarts(needed_code_sites);
+    emp::vector<int> true_useful_starts = {0, 6, needed_code_sites_c.size() - 5,
```

Comment:
This line is causing a 'narrowing conversion' error on my compiler and GitHubs, so you should fix it, specifically the problem is the 'needed_code_sites_c.size() -5' "error: non-constant-expression cannot be narrowed from type
      'unsigned long' to 'int' in initializer list"